lib/commands/serve.js

```diff
@@ -37,13 +38,41 @@ module.exports.builder = extend(
 module.exports.handler = function serve(argv) {
   argv._handled = true;
   var server = new Server(argv.port);
+  var watcher;
+
   server.on('listening', function () {
-    process.stdout.write('documentation.js serving on port ' + argv.port + '\n');
+    process.stdout.write(`documentation.js serving on port ${argv.port}\n`);
   });
-  build(extend({}, { format: 'html' }, argv), function (err, output) {
-    if (err) {
-      return server.setFiles([errorPage(err)]).start();
+
+  function updateWatcher() {
+    if (!watcher) {
+      watcher = chokidar.watch(argv.input);
+      watcher.on('all', debounce(updateServer, 300));
     }
-    server.setFiles(output).start();
-  });
+
+    documentation.expandInputs(argv.input, argv)
+      .then(files => {
+        watcher.add(files.map(data =>
+          typeof data === 'string' ? data : data.file));
+      }, err => {
```

Comment:
General advice is to use `then(a).catch(b)` instead of `then(a, b)` since the former would catch errors in a too.